mozilla/activity-stream
# Mirror directories between mc / github

This PR is intended to make our github repository closer to a mirror of the browser/extensions/activity-stream/ folder in MC. It's based off @dmose's work in https://github.com/mozilla/activity-stream/pull/4160 with a slightly different approach:

- Instead of moving files in mc to a `system-addon/` subdirectory, I chose to move files in our github repository. The reason for this is that all the code that needs to be changed in mc for file moves (such as eslint, license files) are more distributed whereas all the changes to get rid of `system-addon/` in github are local to our repo.
- I added a `bundle` task that could be run from the mozilla central directory directly, `buildmc` still works as it does now
- Syncing is done with `rsync` and a `.mcignore` file listing all files and directories to ignore
- I wasn't sure what to do with the startmc copy watching task, I kept `cpx` with a glob there but having that in two places is not great

The first commit here is a directory move commit, the second contains all path changes and new build tasks.

To test this, I would recommend:
- Run buildmc from the github directory and see if the result runs correctly from the `mozilla-central` directory;
- Make a change in browser/extensions/activity-stream/ and run `npm install && npm run bundle && ./mach build browser/extensions/activity-stream && ./mach run` to see if your change was compiled correctly
- Run` importmc` from github directory to see if the change was correctly imported
